pico-slack.js

```diff
@@ -7,7 +7,7 @@ let pingTimer;
 const startPing = (id = 0)=>{
 	if(!Slack.connected) return;
 	Slack.socket.send(JSON.stringify({ id, type : 'ping' }));
-	pingTimer = setTimeout(()=>startPing(id++), 5000);
+	pingTimer = setTimeout(()=>startPing(++id), 5000);
```

Comment:
I see the commit message for this is "Fix ping IDs failing to increment." How does this fix that? Doesn't this just mean that the ping IDs start at `id + 1` instead of `id`?

Reply:
No, the ID still starts at 0 since the first time it's called is without any args.  Which side of the variable you place the increment operator determines whether the value is read & passed through before or after it's actually incremented.

```javascript
x == 0; x++ == 0; x == 1;
y == 0; ++y == 1; y == 1;
```

This variable doesn't actually need to be incremented because it's not reused inside a closure, it's just a new argument to a new invocation of the function each time.  So every time `startPing` was being called, it would pass in 0 and _then_ increment the arg, but the next invocation already received the 0, so it passes on 0, which passes on 0, etc.

Ultimately I don't think this actually affects anything.  I ran some ping tests and Slack doesn't seem to care if you re-use the same ID.  The only purpose for it as far as I can tell is for the client to be able to identify which "requests" Slack is responding to asynchronously over websocket.  Since these pings are the only outgoing websocket messages we're sending, and we don't care about the responses, it doesn't really matter.  But I still figured it doesn't hurt, just in case things change.

Reply:
Ahhh very good, I hadn't noticed that it's a self-reference. I'm actually thinking that it would be clearer to pass in `i + 1`, since incrementing the variable is strictly not necessary.